### ISSUE TITLE
UI Bug with Files?

### ISSUE THREAD

**Issue by lemongrabber** [ConfigHubPub/ConfigHubPlatform#24]
After creating a new file, we noticed that that file becomes the only one visible from the UI (though all of the other files are still there, and will show up when you start searching for them in the search bar). When inspecting the revision history we noticed this "Path" change along with the creation of the new file.
<img width="1181" alt="screen shot 2017-06-09 at 11 34 51 am" src="https://user-images.githubusercontent.com/6097892/26995425-2a1af562-4d32-11e7-9fa2-d65b855f8cf3.png">

**Comment by edoarsla** [comment]
You're running v1.4?  

Just to clarify, sounds like you're describing 2 problems:

1. After you create a new file, the files page hides all other files for you, only showing the one you just created.

2. On the revisions page (screenshot you included) "changed: " is displayed, which obviously it should not be as this is a new file.  We'll make this wording better.

If I understand this correctly, #1 is the concern.  I tried to reproduce this on v1.4, and have not been able to do it.  Is this something you can reproduce consistently?  If so, could you shed any more light as to how I could.

Thanks,
Edin

**Comment by lemongrabber** [comment]
I was just assuming that the issues were related. In one instance we created a new file, and in another instance we changed the name of an existing file. Something that might be important to note is that new files created after this initial file are visible as well. And I think (but am not 100% sure) that the new file which caused this issue was the first new file we created after upgrading to 1.4

**Comment by edoarsla** [comment]
Just a few questions:

1. Are you able to navigate to the files page, and see all files you expect to see?
2. If not, could you see expected files using a different browser / computer (to eliminate caching, or browser storage issues)
3. If you are able to see all files - can you reproduce the problem by creating a new file, and then not being able to see files you expect to see?

**Comment by edoarsla** [comment]
And just for a good measure - the revisions page (your screenshot), its an unrelated issue - messy UI which we will clean up.

**Comment by lemongrabber** [comment]
1. No, the files page only shows the files created since the first new files was created after we upgraded to 1.4
2. Both Jordan and I have experienced the same issue on different computers, with different browsers (Firefox and Chrome)
3. New files created after the initial issue can be seen fine. If it would be helpful for your debugging, I could send over a dump of our database

**Comment by edoarsla** [comment]
Yes. If you can do that, it would go a long way. You know my email.

**Comment by edoarsla** [comment]
So, the fix when upgrading from v1.3.3 to anything later is:

```
UPDATE AbsoluteFilePath SET path = CASE path WHEN '' THEN NULL ELSE path END;
ALTER TABLE AbsoluteFilePath DROP COLUMN e;
```

Run this on your db.main instance, and restart ConfigHub instance afterwards.

**Comment by lemongrabber** [comment]
We recently ran into a similar issue when adding a new file into our repo, except this time, the new file is the one that will not show up in the default view, but will appear when searched for. Running the first database script (the UPDATE statement) seemed to fix it (The second command failed because that column was no longer present)

**Comment by edoarsla** [comment]
Thanks for letting me know.  We'll take a look.

**Comment by edoarsla** [comment]
We addressed this in v1.5.
https://github.com/ConfigHubPub/ConfigHubPlatform/releases/tag/v1.5
